### PR TITLE
Move coverage to GitHub actions

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `ENOMEM: not enough memory, stat /home/circleci/babel/packages/just/click/the/rerun/button`
| Tests Added + Pass?      | Yes
| License                  | MIT

This PR moves the coverage report job to GitHub Actions. The ENOMEM error should be alleviated because

| | CPU | RAM |
| --- | --- | --- |
| [GitHub](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources) | 2 core | 7GB |
| [CircleCI](https://circleci.com/docs/2.0/configuration-reference/#docker-executor) | 2 core | 4GB |

Merge tips:
Please add `CODECOV_TOKEN` to the `Settings > Secrets` after merging to master. See https://github.com/codecov/codecov-action#usage

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Generate coverage report
+        run: |
+          yarn --version
+          make test-ci-coverage
+          # Builds babel-standalone with the regular Babel config
+          # test-ci-coverage doesn't test babel-standalone, as trying to gather coverage
+          IS_PUBLISH=true make build-standalone
+          # data for a JS file that's several megabytes large is bound to fail. Here,
+          # we just run the babel-standalone test separately.
+          yarn jest "\-standalone/test"
```
Comment (nicolo-ribaudo) [inline .github/workflows/coverage.yml]: Since this is just for coverage, maybe we can avoid testing standalone here?

Comment (JLHwung) [reply .github/workflows/coverage.yml]: The standalone is only tested on the coverage job in the CI, we have skipped the test on other executors.

Comment (nicolo-ribaudo) [reply .github/workflows/coverage.yml]: Oh then maybe we should test it separately? I don't really see how it fits in the coverage tests.

Comment (JLHwung) [reply .github/workflows/coverage.yml]: Yeah but standalone will rely on building and installing all the dependencies, I think we can test it here to save redundant machine hours.

Comment (nicolo-ribaudo) [reply .github/workflows/coverage.yml]: Ok fair

**Pull request merged**